src/main/scala/thurloe/dataaccess/HttpSendGridDAO.scala

```diff
@@ -0,0 +1,81 @@
+package thurloe.dataaccess
+
+import com.sendgrid.SendGrid.Response
+import com.sendgrid._
+import com.typesafe.config.ConfigFactory
+import thurloe.database.ThurloeDatabaseConnector
+import thurloe.service.Notification
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+ * Created by mbemis on 6/16/16.
+ */
+class HttpSendGridDAO {
+
+  val configFile = ConfigFactory.load()
+  val sendGridConfig = configFile.getConfig("sendgrid")
+  val apiKey = sendGridConfig.getString("apiKey")
+  val substitutionChar = sendGridConfig.getString("substitutionChar")
+  val fromAddress = sendGridConfig.getString("defaultFromAddress")
+
+  val dataAccess = ThurloeDatabaseConnector
+
+  def sendNotification(notification: Notification): Future[Response] = {
+    lookupPreferredEmail(notification.userId) flatMap { preferredEmail =>
+      val email = createEmail(preferredEmail, notification.notificationId, notification.substitutions)
+      sendEmail(email)
+    }
+  }
+
+  /*
+    Note: email.setSubject and email.setText must be set even if their values
+    aren't used. Supposedly this will be fixed in a future version of SendGrid
+   */
+  def createEmail(toAddress: String, notificationId: String, substitutions: Map[String, String] = Map.empty): SendGrid.Email = {
+    val email = new SendGrid.Email()
+
+    email.addTo(toAddress)
+    email.setFrom(fromAddress)
+    email.setTemplateId(notificationId)
+    email.setSubject(" ")
+    email.setText(" ")
+    addSubstitutions(email, substitutions)
+  }
+
+  def sendEmail(email: SendGrid.Email): Future[Response] = {
+    val sendGrid = new SendGrid(apiKey)
+
+    Future {
+      val response = sendGrid.send(email)
```

Comment:
who is responsible for retrying this call in the event of a failure? do we retry here or is the consumer responsible for retrying?